prov/gni/configure.m4

```diff
@@ -40,32 +40,36 @@ AC_DEFUN([FI_GNI_CONFIGURE],[
 	       ])
 
 	have_criterion=false
+	criterion_tests_present=true
 
-	AC_ARG_WITH([criterion],
-		[AS_HELP_STRING([--with-criterion],
-			       [Location for criterion unit testing framework])])
+	AS_IF([test -d $srcdir/prov/gni/test],
+	      [AC_ARG_WITH([criterion], [AS_HELP_STRING([--with-criterion],
+		       [Location for criterion unit testing framework])])],
+		 	[criterion_tests_present=false])
```

Comment:
Putting an `AC_ARG_WITH` inside of an `AS_IF` (i.e., inside of a shell `if` construct) is pretty misleading because of the way that `AC_ARG_WITH` is implemented by autoconf.  When autoconf is processing the `configure.ac` file, it notices the `AC_ARG_WITH` macro and uses m4 "diversions" hoist most of the option processing up close to the start of the resulting `configure` script.  All that remains is a check against whether the `$with_X` parameter ended up being  set by the hoisted logic.  For example, the snippet above expands to this in the current version of the `master` branch (additional lines shown for context):
```sh
        have_criterion=false
        criterion_tests_present=true

        if test -d $srcdir/prov/gni/test; then :

# Check whether --with-criterion was given.
if test "${with_criterion+set}" = set; then :
  withval=$with_criterion;
fi

else
  criterion_tests_present=false
fi

        if test "$with_criterion" != "" && test "$with_criterion" != "no"; then
[...]
```

The output of `./configure --help` will always show `--with-criterion`, whether or not the `prov/gni/test` dir is present, even though you were probably trying to avoid that.  The above bit of `configure.m4` should probably just be this instead (untested):

```
        have_criterion=false

        AC_ARG_WITH([criterion],
                    [AS_HELP_STRING([--with-criterion],
                                    [Location for criterion unit testing framework])])
```

And then change this line below:
```
                AS_IF([test "$criterion_tests_present" = "true"],
```
Into this line instead:
```
                AS_IF([test -d $srcdir/prov/gni/test],
```
eliminating the `$criterion_tests_present` variable altogether.